### PR TITLE
Add support for 22.2 protocol buffer version.

### PR DESCRIPTION
**Pull request by kylepl**
Add support for 22.2 protocol buffer version. This requires a number of things, I welcome feedback:

* Updating upb to have a later version and depend on it (a rule now depends on a later version)
* Since `utf_range` is not in bzlmod as itself, but is within the library, re-direct references to it to be internal references. This is a feature added since the last version.
* Disable Ruby support - since it is not feasible currently in a bzlmod only world (rules_ruby is not on bzlmod).
* This removes one patch for using local aliases, since it was changed upstream.

### CODE REVIEW

**Comment by meteorcloudy** [comment]
```
patching file MODULE.bazel
patching file examples/MODULE.bazel
patching file conformance/conformance_test_runner.sh
patching file conformance/failure_list_java.txt
Reversed (or previously applied) patch detec
```
The patch file seems to have some problem. (But the pipeline should fail instead of getting blocked, I thought I had fixed that...)

**Comment by malt3** [comment]
The file encoding seems to have changed:

```
file protobuf-2*/conformance/failure_list_java.txt
protobuf-21.7/conformance/failure_list_java.txt: ASCII text, with CRLF line terminators
protobuf-22.2/conformance/failure_list_java.txt: ASCII text
```

File: modules/protobuf/22.2/patches/add_module_dot_bazel.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 0000000..140f322
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,42 @@
++module(
++    name = "protobuf",
++    compatibility_level = 1,
++    version = "21.7",
```
Comment (malt3) [inline modules/protobuf/22.2/patches/add_module_dot_bazel.patch:16]:
```suggestion
+    version = "22.2",
```

File: modules/protobuf/22.2/patches/add_missing_files.patch
Comment (malt3) [inline modules/protobuf/22.2/patches/add_missing_files.patch:1]: No longer needed. The zip archive for releases >= 22.0 already ships the files created by this diff.

File: modules/protobuf/22.2/patches/add_module_dot_bazel.patch [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
++
++bazel_dep(name = "bazel_skylib", version = "1.0.3")
++bazel_dep(name = "rules_python", version = "0.10.2")
++bazel_dep(name = "rules_cc", version = "0.0.1")
++bazel_dep(name = "rules_proto", version = "4.0.0")
++bazel_dep(name = "rules_java", version = "4.0.0")
++bazel_dep(name = "rules_pkg", version = "0.7.0")
++bazel_dep(name = "abseil-cpp", repo_name = "com_google_abseil", version = "20211102.0")
++bazel_dep(name = "zlib", version = "1.2.11")
++bazel_dep(name = "upb", version = "0.0.0-20220923-a547704")
```
Comment (malt3) [inline modules/protobuf/22.2/patches/add_module_dot_bazel.patch:27]:
```suggestion
+bazel_dep(name = "upb", version = "0.0.0-20230320-666a28e")
```

File: modules/protobuf/22.2/patches/add_missing_files.patch
Comment (kylepl) [reply modules/protobuf/22.2/patches/add_missing_files.patch:1]: I have forgotten the context of this, so I've removed it for now.

**Comment by kylepl** [comment]
I generally forgot my state of when I was trying to do this before, but I've responded to the few comments. 

I think the changing of file formats should be gone now since I've removed that diff, but I forgot why I thought I needed it.

**Comment by Wyverald** [comment]
Hello! I'm doing some routine cleanup of stale PRs. If you're still working on this and planning to make progress soon, please let me know and we can re-open it.

**Pull request closed without merging**